packages/block-editor/src/components/list-view/block.js

```diff
@@ -68,6 +68,15 @@ function ListViewBlock( {
 		level
 	);
 
+	const isLocked = useSelect(
+		( select ) => {
+			const { canMoveBlock, canRemoveBlock } = select( blockEditorStore );
+
+			return ! canMoveBlock( clientId ) || ! canRemoveBlock( clientId );
+		},
+		[ clientId ]
+	);
```

Comment:
I should probably extract this into a hook. Now there're three places where we need similar data.

Example hook:

```js
const { canMove, canRemove, canLockBlock } = useBlockLock( clientId: string, rootClientId?: string ); 
````